### PR TITLE
Add softmax related loss functions

### PR DESCRIPTION
**Pull request by KevinXuxuxu**
Added 2 softmax related loss function
- logsoftmax from [deeplearningbook 6.2.2.3](https://www.deeplearningbook.org/contents/mlp.html)
- cross entropy with softmax from [deepnotes.io](https://deepnotes.io/softmax-crossentropy#cross-entropy-loss)

Also adjusted layer initialization thanks to suggestion from @luyueheng

### CODE REVIEW

**Comment by KevinXuxuxu** [comment]
I'm hoping https://github.com/KevinXuxuxu/NN/pull/7 to be merged first so that I can add better result to mnist example.

**Review by luyueheng — APPROVED** [review]
Can add softmax as activation function later
